flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendTestBase.java

```diff
@@ -460,22 +461,66 @@ public void testGetKeysAndNamespaces() throws Exception {
                     backend.getPartitionedState(
                             ns1,
                             StringSerializer.INSTANCE,
-                            new ValueStateDescriptor<>(fieldName, IntSerializer.INSTANCE));
+                            new ValueStateDescriptor<>(
+                                    valueStateFieldName, IntSerializer.INSTANCE));
+
+            MapState<Integer, Integer> keyedState2 =
```

Comment:
I think modify existing test is okay, however, I don't we should introduce so much code duplication.
I think you could leverage `backend.createInternalState` API which could set current namespace of `ns1` or `ns2`.

Reply:
OK , I noticed that there is a pair of unit tests `#testGetKeys` and `#testMapStateGetKeys` in `StateBackendTestBase`, which is very similar to my situation. I referred to these codes when I modified it before.

Reply:
In addition, I observed that similar tests in StateBackendTestBase seem to have some duplicate codes.(eg `testListStateRestoreWithWrongSerializers`\ `testValueStateRestoreWithWrongSerializers` `testMapStateRestoreWithWrongSerializers`)
So maybe it's also fine to add a new test as `#testMapStateGetKeysAndNamespace` just like` #testMapStateGetKeys ` or do u have some other greater suggestions ?

Reply:
I think you could add a new `#testMapStateGetKeysAndNamespace` method.

Reply:
Thanks for your advice, have resolved it .:)